include/clap/ext/draft/undo-redo.h

```diff
@@ -0,0 +1,87 @@
+#pragma once
+
+#include "../../plugin.h"
+#include "../../string-sizes.h"
+#include "../../stream.h"
+
+/// @page Undo/Redo

```

Comment:
A few things that are missing

1. We show the undo/redo stack in our UI. Can we query the host undo/redo stack here?

3. The host will put *other* things on the undo redo stack. Do we want to be able to trigger them from the ui and interleave those stacks? I'm thinking especially of things like a setting a parameter from outside the plugin - what does that do as far as undo/redo? We really need to document that case.

Reply:
My approach here would be that the plugin only knows about its internal undo/redo stacks and not the host's stacks. So if the plugin does internal undo/redo, the host might remove those steps from between own undo/redo steps. I think this would simplify things for the plugin devs and the host would be responsible for placing the plugin undo/redo objects in the appropriate place. You think that's OK?

Concerning parameter changes from outside, I would add a note to the documentation that the plugin is always responsible for creating undo steps for parameter changes, even if they come from the host and the host must not create undo steps for those parameter changes (?)